phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -326,6 +326,49 @@ private IndexMaintainer(PTable dataTable, PTable index) {
         // TODO: check whether index is immutable or not. Currently it's always false so checking
         // data table is with immutable rows or not.
         this.immutableRows = dataTable.isImmutableRows();
+        int indexColByteSize = 0;
+        for (int i = indexPosOffset; i < index.getPKColumns().size(); i++) {
+            PColumn indexColumn = index.getPKColumns().get(i);
+            int indexPos = i - indexPosOffset;
+            Expression expression = null;
+            try {
+                ParseNode parseNode  = SQLParser.parseCondition(indexColumn.getExpressionStr());
```

Comment:
Will indexColumn.getExpressionStr() be null for a non functional index? If so, instead of the instanceof check for ColumnExpression below, can we do that here based on the null check?